docs/docsite/rst/network_best_practices_2.5.rst

```diff
@@ -0,0 +1,372 @@
+         register: backup_eos_location
+         when: ansible_network_os == 'eos'
+
+       - name: backup switch (vyos)
+         vyos_config:
+           backup: yes
+         register: backup_vyos_location
+         when: ansible_network_os == 'vyos'
+
+       - name: Create backup dir
+         file:
+           path: "/tmp/backups/{{ inventory_hostname }}"
+           state: directory
+           recurse: yes
+
+       - name: Copy backup files into /tmp/backups/ (eos)
+         copy:
+           src: "{{ backup_eos_location.backup_path }}"
+           dest: "/tmp/backups/{{ inventory_hostname }}/{{ inventory_hostname }}.bck"
+         when: ansible_network_os == 'eos'
+
+       - name: Copy backup files into /tmp/backups/ (vyos)
+         copy:
+           src: "{{ backup_vyos_location.backup_path }}"
+           dest: "/tmp/backups/{{ inventory_hostname }}/{{ inventory_hostname }}.bck"
+         when: ansible_network_os == 'vyos'
+
+Running the playbook
+--------------------
+
+To run the playbook, run the following from a console prompt:
+
+.. code-block:: console
+
+   ansible-playbook -i inventory facts-demo.yml
+
+This should return output similar to the following:
+
+.. code-block:: console
+
+   PLAY RECAP
+   eos01.example.net          : ok=7    changed=2    unreachable=0    failed=0
+   ios01.example.net          : ok=7    changed=2    unreachable=0    failed=0
+   vyos01.example.net         : ok=6    changed=2    unreachable=0    failed=0
+
+Next, look at the contents of the file we created containing the switch facts:
+
+.. code-block:: console
+
+   cat /tmp/switch-facts
+
+You can also look at the backup files:
+
+.. code-block:: console
+
+   find /tmp/backups
+
+
+If `ansible-playbook` fails, please follow the debug steps in :doc:`network_debug_troubleshooting`.
+
+
+Implementation Notes
+====================
+
+
+Demo variables
+--------------
+
+Although these tasks are not needed to write data to disk, they are used in this example to demonstrate some methods of accessing facts about the given devices or a named host.
+
+Ansible ``hostvars`` allows you to access variables from a named host. Without this we would return the details for the current host, rather than the named host.
+
+For more information, see :ref:`magic_variables_and_hostvars`.
+
+Get running configuration
+-------------------------
+
+The :ref:`eos_config <eos_config>` and :ref:`vyos_config <vyos_config>` modules have a ``backup:`` option that when set will cause the module to create a full backup of the current ``running-config`` from the remote device before any changes are made. The backup file is written to the ``backup`` folder in the playbook root directory. If the directory does not exist, it is created.
+
+To demonstrate how we can move the backup file to a different location we ``register`` the result and use the ``backup_path`` return value as source location to move the file into ``/tmp/backups/`` directory which we have created.
```

Comment:
Hard to parse. If I'm understanding it correctly, maybe try: "To demonstrate how we can move the backup file to a different location, we ``register`` the result and move the file to the path stored in ``backup_path``."